apps/api/src/app/ai/usecases/generate-content/generate-content.usecase.ts

```
@@ -0,0 +1,264 @@
+- Avoid special characters that might not render properly`,
+      [StepTypeEnum.PUSH]: `
+Push notification guidelines:
+- Title should be under 50 characters
+- Body should be under 150 characters
+- Create urgency or value proposition
+- Be specific about what the user should do`,
+      [StepTypeEnum.IN_APP]: `
+In-app notification guidelines:
+- Can be slightly longer than push notifications
+- Include action buttons when appropriate
+- Be contextual to the user's current state
+- Use clear, actionable language`,
+      [StepTypeEnum.CHAT]: `
+Chat message guidelines:
+- Keep messages conversational
+- Be friendly but professional
+- Include relevant context
+- Make it easy to respond or take action`,
+    };
+
+    let prompt = basePrompt + (channelGuidelines[stepType] || '');
+
+    if (context) {
+      prompt += '\n\nContext:';
+      if (context.workflowName) {
+        prompt += `\n- Workflow: ${context.workflowName}`;
+      }
+      if (context.workflowDescription) {
+        prompt += `\n- Purpose: ${context.workflowDescription}`;
+      }
+      if (context.variables && context.variables.length > 0) {
+        prompt += `\n- Available variables: ${context.variables.join(', ')}`;
+      }
+    }
+
+    return prompt;
+  }
+
+  async execute(command: GenerateContentCommand): Promise<GenerateContentResponseDto> {
+    this.logger.info(`Generating ${command.stepType} content for user: ${command.userId}`);
+
+    const schema = this.getSchemaForStepType(command.stepType);
+    const systemPrompt = this.getSystemPrompt(command.stepType, command.context);
+
+    const result = await generateObject({
+      model: openai('gpt-4o'),
+      schema,
+      system: systemPrompt,
+      messages: command.messages.map((msg) => ({
+        role: msg.role,
+        content: msg.content,
+      })),
+    });
+
+    this.logger.info(`Successfully generated ${command.stepType} content`);
+
+    const response = result.object as GenerateContentResponseDto;
```

should we parse the result to make sure that the response is a proper type response.content GeneratedContent? in case that the generation failed to follow the structure, we need to throw an exception, otherwise the client will behave unexpectedly

---

Sure, Let me take care of this